js/interactive-guides/circuit-breaker/circuit-breaker-callback.js

```diff
@@ -313,34 +313,56 @@ var circuitBreakerCallBack = (function() {
     };
 
     var __listenToEditorForCircuitBreakerAnnotationChanges = function(editor){
-
-        var __showCircuitBreakerInPod = function(){
+        var __listenToContentChanges = function(editorInstance, changes) {
             // Get pod from contentManager
-            var cb = contentManager.getCircuitBreaker(editor.getStepName());
-
+            var cb = contentManager.getCircuitBreaker(editor.getStepName());            
             // Get the parameters from the editor and send to the circuitBreaker
             var content = editor.getEditorContent();
             try{
             var annotation = content.match(/@CircuitBreaker(.|\n)*\)/g)[0];
-              annotation = annotation.substring(0,annotation.indexOf("public")).trim(); // Get rid of the public Service...
```

Comment:
The regex on line 322 could change to "public class BankService {\s*@CircuitBreaker\s*\(((\s|\S)*?)\)\s*public Service checkBalance" and group#1 will capture the parameters. You cannot use content.match anymore. Instead you've to use:
var matchPattern = "public class BankService {\s*@CircuitBreaker\s*\(((\s|\S)*?)\)\s*public Service checkBalance";
var regexToMatch = new RegExp(matchPattern, "g");
var groups = regexToMatch.exe(content);
var params = groups[1];

Reply:
Fixed